src/types/types.ts

```diff
@@ -18,17 +18,24 @@ export type IPCFolder = {
 	logs: FileLog[];
 };
 
+export type ProgramLog = {
+	action: string;
+	date: number;
+};
+
 export type FileLog = {
 	action: string;
 	date: number;
 };
 
 export type IPCProgram = {
+	id: string;
 	hash: string;
 	name: string;
 	createdAt: number;
 	entrypoint: string;
 	size: number;
+	log: ProgramLog[];
```

Comment:
```suggestion
	logs: ProgramLog[];
```
An array of log, adding an `s` looks better